### ISSUE TITLE
Not abble to launch squirel.py on ubuntu 20.04

### ISSUE THREAD

**Issue by blairfancy** [julesontheroad/NSC_BUILDER#163]
I try to run 
python3 squirrel.py -lib_call Interface start

but i got 

error:Exception: name 'slimpath' is not defined

any advice to make it works?

**Comment by julesontheroad** [comment]
Is a fallback, it checks for chromium, then chromium portable in the chromium folder and lastly slimjet portable, it's not finding chrome or chromium I think. 

If you don't have chromium there's 2 easy steps:

**Server mode, for this take my current master since it has some fixes:**

python3 squirrel.py -lib_call Interface server -xarg portnumber host
So for example for port 9001 and host localhost:

**python3 squirrel.py -lib_call Interface server -xarg 9001 localhost**

or since localhost is the default:

**python3 squirrel.py -lib_call Interface server -xarg 9001**

If you want to use your ip + localhost:

**python3 squirrel.py -lib_call Interface server -xarg 9001 0.0.0.0**

-xarg are extended arguments, can also be decorated like this, and it will  just use what's after **=**.  Arguments need to be ordered and you can only skip arguments after the last you input since there's no assignation implemented in -lib_call

**python3 squirrel.py -lib_call Interface server -xarg "port=9001" "host=localhost"**

**For interface this will probably work:**

**python3 squirrel.py -lib_call Interface start -xarg "browserpath=default"**

 If you want multiconsole interface you'll have to do:

**python3 squirrel.py -lib_call Interface start -xarg "browserpath=default" videoplayback=True height=800 width=740 port=rg8000** 

rg8000 asigns each new console window an open port between 8000 and 8999.

For no console + multiwindow:
**python3 squirrel.py -lib_call Interface start -xarg "browserpath=default" videoplayback=True height=800 width=740 port=rg8000 "host=localhost" "noconsole=True"** 

Still maybe i should add a preset with multiwindow and other with multiwindow + noconsole.

**Comment by blairfancy** [comment]
Stange because chromium is installed in my case. Used `snap install chromium `
chromium is in /snap/bin folder.

**Comment by julesontheroad** [comment]
Hmm i'll have to check if the scrypt finds chromium fine in linux. I have to set up a virtual box for that since my current linux server is headless. When i used one that also had vnc allowed i know it work but i don't remember if i specified the path.
I think adding the absolute path should also work:

**python3 squirrel.py -lib_call Interface start -xarg " /snap/bin/chromium"**

**Comment by blairfancy** [comment]
nop i got slimpath error when i send full path.

Is there any interactive mode like with .bat?

**Comment by blairfancy** [comment]
after removing space in command i got error :
Exception: module '_EEL_' has no attribute 'browsers'

**Comment by blairfancy** [comment]
Sory it s probably not a good place to ask question but is there any option to convert XCI to NSP i one step ?

**Comment by Masamune3210** [comment]
To the last question, just use the first option with the xci or folder of xci as input and build as nsp

**Comment by blairfancy** [comment]
i cant use bat file on linux how to do the same with squirel.py?

**Comment by julesontheroad** [comment]
He means on linux:

**squirrel.py -dc "filepath" -o "output_folder" -t "nsp"** is the more basic command, it will also do xcz to nsp and nsz to ticketless nsp. For not ticketless you need to use either decompressor or rebuild.

About the linux thing i'll have to check I had it running before with Interface but i may had modified the Interface.py file, i don't remember.

Anyway the server option works since i used it to stream the Interface from a remote headless server directly both with the bottle server it has integrated and also redirecting it with ngrok. 
I'll take a look when i can, anyway **are you using current master?** That seems like this bug i fixed:
https://github.com/julesontheroad/NSC_BUILDER/commit/5ec69be005783274cb2ea495b3b684e1e307f1c8

Basically i forked eel and added it as a local package called _EEL_ to add the server stuff and allow launching it with a hidden console among other stuff. I forgot to rename those calls in the package so the previous release had that bug if you didn't also had eel installed. It's fixed on the master.

**Comment by blairfancy** [comment]
python3 squirrel.py -lib_call Interface server -xarg 9001 return error:
Exception: local variable 'flag_file' referenced before assignment

**Comment by blairfancy** [comment]
With NSC 099 command python3 squirrel.py -lib_call Interface start -xarg "browserpath=default" return:
```
lib/Config.py:180: SyntaxWarning: list indices must be integers or slices, not str; perhaps you missed a comma?
  paths.titleDatabase = ['paths']['titledb']
Getting NUTDB json
Launched using default system browser
Exception: [Errno 2] No such file or directory: '/home/blairfancy/Bureau/switch/dump_cart/NSB_099py/zconfig/local_libraries.txt'
Exception: [Errno 2] No such file or directory: '/home/blairfancy/Bureau/switch/dump_cart/NSB_099py/zconfig/remote_libraries.txt'
Traceback (most recent call last):
  File "/home/blairfancy/Bureau/switch/dump_cart/NSB_099py/ztools/_EEL_/__init__.py", line 304, in _process_message
    return_val = _exposed_functions[message['name']](*message['args'])
  File "lib/Interface.py", line 488, in getfname
    root = tk.Tk()
NameError: name 'tk' is not defined
```

**Comment by julesontheroad** [comment]
ok, several stuff. 
1.- First use the master, the flag error is other fix. If the folder where the Interface logs are stored is not created it gives that error. I'll release the exes soon but since you're using on linux i'd advice to always use the master.
2.- The tk error is because when you built your python 3 you didn't built it with tkinter. Tkinter is used to load files by the interface.
In normal ubuntu it seems you can get it easilly -> 
**sudo apt-get install python3-tk**
https://riptutorial.com/tkinter/example/3206/installation-or-setup
If you need to compile a python version it gets a little more tricky since you needed a dev tools package, i don't exactly remember the steps right now. The thing is normal python distribution doesn't have tkinter in linux built in even if it has on windows.

**Comment by Masamune3210** [comment]
My bad, spaced on the whole "bat files are a completely windows thing"

**Comment by julesontheroad** [comment]
@Masamune3210 I think he meant that i wrote **" /snap**/bin/chromium" instead of  **"/**snap/bin/chromium". First time path wasn't correct so it detected as non existing path and trigger auto. Auto didn't locate chromium or chromium portable and the 3rd alternative was slimjet portable.
Then when the path was correct he ran into the bug i fixed on https://github.com/julesontheroad/NSC_BUILDER/commit/5ec69be005783274cb2ea495b3b684e1e307f1c8 since it has to called that file to load the browser.

**Comment by julesontheroad** [comment]
The fix is also included with the new release, since https://github.com/julesontheroad/NSC_BUILDER/issues/168 refers to the specific interface issue i-m closing this, leaving the other issue open till confirmation.
